lambda/simplif.ml

```diff
@@ -40,8 +40,12 @@ let rec eliminate_ref id = function
   | Lmutlet(kind, v, e1, e2) ->
       Lmutlet(kind, v, eliminate_ref id e1, eliminate_ref id e2)
   | Lletrec(idel, e2) ->
-      Lletrec(List.map (fun (v, clas, e) -> (v, clas, eliminate_ref id e)) idel,
-              eliminate_ref id e2)
+      let bindings =
+        List.map (fun { id = v; rkind; def } ->
+            { id = v; rkind; def = eliminate_ref id def })
```

Comment:
I think that this is `(fun r -> { r with def = eliminate_ref id r.def })`, which avoids the subtle issue of not shadowing `id` when opening the record.